pandas/core/arrays/datetimes.py

```diff
@@ -324,6 +430,26 @@ def _generate_range(cls, start, end, periods, freq, tz=None,
 
         return cls._simple_new(index.asi8, freq=freq, tz=tz)
 
+    # -----------------------------------------------------------------
+    # DatetimeLike Interface
+
+    def _unbox_scalar(self, value):
+        if not isinstance(value, (self._scalar_type, type(NaT))):
+            raise ValueError("'value' should be a a Timestamp..")
+        return value.value
+
+    def _scalar_from_string(self, value):
+        return Timestamp(value, tz=self.tz)
+
+    def _check_compatible_with(self, other):
+        if not timezones.tz_compare(self.tz, other.tz):
```

Comment:
same comment

Reply:
Which comment is that?

Reply:
about doc-strings

Reply:
The base class has the docstring. I've expanded the one for `_check_comatible_with` to say what each subclass does (not pushed yet).

Reply:
This has been pushed now. For reference:

```python
    def _check_compatible_with(self, other):
        """
        Verify that `self` and `other` are compatible.

        * DatetimeArray verifies that the timezones (if any) match
        * PeriodArray verifies that the freq matches
        * Timedelta has no verification

        Parameters
        ----------
        other

        Raises
        ------
        Exception
        """
 ```